hack/ci/e2e-olm.sh

```diff
@@ -82,30 +177,212 @@ function check_spo_is_running() {
+
+    return 0
+}
+
+function smoke_test_multi() {
+    kubectl create ns sp-test-1
+    sp_in_ns sp-test-1
+    kubectl wait --for=condition=ready -nsp-test-1 sp log-all || return 1
+
+    kubectl create ns sp-test-2
+    sp_in_ns sp-test-2
+    kubectl wait --for=condition=ready -nsp-test-2 sp log-all || return 1
+
+    # SPO always adds its own ns regardless even if not watched explicitly
+    sp_in_ns security-profiles-operator
+    kubectl wait --for=condition=ready -nsecurity-profiles-operator sp log-all || return 1
+
+    # negative test, we listen for sp-test-{1,2} only
+    kubectl create ns sp-test-3
+    sp_in_ns sp-test-3
+    kubectl wait --for=condition=ready -nsp-test-3 sp log-all && return 1
+
+    kubectl delete sp --all --all-namespaces
+    kubectl delete ns sp-test-{1,2,3}
+
+    # in this installation method, the CSV is installed into
+    # security-profiles-operator ns only
+    assert_spo_csv_installed_in_ns security-profiles-operator || return 1
+    assert_spo_csv_copied_to default security-profiles-operator && return 1
+
+    return 0
+}
+
+function smoke_test_custom() {
+    kubectl create ns sp-test-1
+    sp_in_ns sp-test-1
+    kubectl wait --for=condition=ready -nsp-test-1 sp log-all || return 1
+
+    kubectl create ns sp-test-2
+    sp_in_ns sp-test-2
+    kubectl wait --for=condition=ready -nsp-test-2 sp log-all || return 1
+
+    sp_in_ns spo-lives-here
+    kubectl wait --for=condition=ready -nspo-lives-here sp log-all || return 1
+
+    kubectl delete sp --all --all-namespaces
+    kubectl delete ns sp-test-1 sp-test-2
+
+    # SPO CSV is installed into the spo-lives-here ns and copied everywhere
+    assert_spo_csv_installed_in_ns spo-lives-here || return 1
+    assert_spo_csv_copied_to default spo-lives-here || return 1
+
+    return 0
+}
+
+function smoke_test() {
+    installation_method=$1
+
+    rv=1 # be pessimistic
+    smoke_test_$installation_method
+    rv=$?
+    echo "Smoke test for $installation_method returned $rv"
+
+    return $rv
+}
+
+function teardown_spo() {
+    installation_method=$1
+    manifests=examples/olm/$installation_method-install-resources.yaml
+
+    # profiles might have finalizers
+    kubectl delete sp --all --all-namespaces
+
+    kubectl delete -f $manifests
 }
 
 # The actual script begins here
 build_and_push_spo
 build_and_push_packages
-deploy_olm
-deploy_spo
-check_spo_is_running
+
+deploy_deps
+
+rv=0
+for method in all own single multi; do
+    echo "Testing SPO deployment in $method namespace(s)"
+
+    deploy_spo $method || rv=1
+    check_spo_is_running security-profiles-operator || rv=1
+    smoke_test $method || rv=1
+
+    teardown_spo $method
+done
+
+# deployment into the custom namespace is a bit special
+echo "Testing SPO deployment in custom namespace(s)"
+deploy_spo_in_custom_ns spo-lives-here || rv=1
+check_spo_is_running spo-lives-here || rv=1
+kubectl get csv -A --show-labels
```

Comment:
hmm, sorry, this is not needed and is just debugging, I can either remove this command if there's comment for this PR or when I submit a patch for the environment variable installations.

Reply:
Let's keep it if it helps debugging.